src/morph/src/Response.re

```diff
@@ -60,86 +62,81 @@ let set_status = (status: Status.t) => {
   result_map(res => {...res, status});
 };
 
-let set_body = (body: body) => {
+let set_body = body => {
   result_map(res => {...res, body});
 };
 
-let ok = (res: t) => {
+let ok = res => {
   add_header(("Content-length", "2"), res)
   |> set_status(`OK)
-  |> set_body(`String("ok"))
-  |> Lwt.return;
+  |> set_body(`String("ok"));
 };
 
-let text = (text, res: t) => {
+let text = (text, res) => {
   let content_length = text |> String.length |> string_of_int;
   add_header(("Content-length", content_length), res)
-  |> set_body(`String(text))
-  |> Lwt.return;
+  |> set_body(`String(text));
 };
 
-let json = (json, res: t) => {
+let json = (json, res) => {
   let content_length = json |> String.length |> string_of_int;
   add_header(("Content-type", "application/json"), res)
   |> add_header(("Content-length", content_length))
-  |> set_body(`String(json))
-  |> Lwt.return;
+  |> set_body(`String(json));
 };
 
-let html = (markup, res: t) => {
+let html = (markup, res) => {
   let content_length = markup |> String.length |> string_of_int;
   add_header(("Content-type", "text/html"), res)
   |> add_header(("Content-length", content_length))
-  |> set_body(`String(markup))
-  |> Lwt.return;
+  |> set_body(`String(markup));
 };
 
-let redirect = (~code=303, targetPath, res: t) => {
+let redirect = (~code=303, targetPath, res) => {
   let content_length = targetPath |> String.length |> string_of_int;
 
   add_header(("Content-length", content_length), res)
   |> add_header(("Location", targetPath))
   |> set_status(`Code(code))
-  |> set_body(`String(targetPath))
-  |> Lwt.return;
+  |> set_body(`String(targetPath));
 };
 
-let unauthorized = (message, res: t) => {
+let unauthorized = (message, res) => {
   add_header(
     ("Content-length", String.length(message) |> string_of_int),
     res,
   )
   |> set_status(`Unauthorized)
-  |> set_body(`String(message))
-  |> Lwt.return;
+  |> set_body(`String(message));
 };
 
-let not_found = (~message="Not found", res: t) => {
+let not_found = (~message="Not found", res) => {
   add_header(
     ("content-length", String.length(message) |> string_of_int),
     res,
   )
   |> set_status(`Not_found)
-  |> set_body(`String(message))
-  |> Lwt.return;
-};
-
-let get_file_stream = file_name => {
-  let read_only_flags = [Unix.O_RDONLY];
-  let read_only_perm = 444;
-  let fd = Unix.openfile(file_name, read_only_flags, read_only_perm);
-  Lwt_io.of_unix_fd(fd, ~mode=Lwt_io.Input) |> Lwt_io.read_chars;
-};
-
-let static = (file_path, res: t) => {
-  Sys.file_exists(file_path)
-    ? {
-      let size = Unix.stat(file_path).st_size;
-      let stream = get_file_stream(file_path);
-      add_header(("Content-type", Magic_mime.lookup(file_path)), res)
-      |> add_header(("Content-length", string_of_int(size)))
-      |> set_body(`Stream(stream))
-      |> Lwt.return;
-    }
-    : not_found(~message="File does not exist", res);
+  |> set_body(`String(message));
 };
+/*
```

Comment:
The static handler was removed when I tried to remove Lwt from core. I think we want to move this to the morph_server package or something where we actually implement the `` `Stream `` body type.
I also think this is the only part in the core of morph that is depending on `Unix`/`Lwt_unix`, if we remove it we're in theory compatible with mirage.

Reply:
From my brief read it does look like a candidate to move out of here indeed.👍

Reply:
This is now moved to morph_server along with the added response body types